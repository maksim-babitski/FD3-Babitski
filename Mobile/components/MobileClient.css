.MobileClient
{
    border-top: 1px solid #95a1ad;
}

.MobileCompany td {
    padding: 8px 0px 8px 25px;
}