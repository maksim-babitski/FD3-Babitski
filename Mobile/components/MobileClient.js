import React from 'react';
import PropTypes from 'prop-types';

import './MobileClient.css';
import {companyEvents} from './events';

class MobileClient extends React.PureComponent {

    static propTypes = {
        info:PropTypes.shape({
            id: PropTypes.number.isRequired,
            lastName:PropTypes.string.isRequired,
            name:PropTypes.string.isRequired,
            middleName:PropTypes.string.isRequired,
            balance: PropTypes.number.isRequired,
        }),
    };

    state = {
        info: this.props.info
    };

    editClient = () => {
        companyEvents.emit("EeditClient",this.props.info.id);
    };

    deleteClient = () => {
        companyEvents.emit("EdeleteClient",this.props.info.id);
    };

    render() {

        console.log("MobileClient id="+this.state.info.id+" render");

        return (
            <tr className='MobileClient'>
                <td>{this.props.info.lastName}</td>
                <td>{this.props.info.name}</td>
                <td>{this.props.info.middleName}</td>
                <td>{this.props.info.balance}</td>
                <td style={{background: this.props.info.balance>0?"green":"red"}}>{this.props.info.balance>0? "active":"blocked"}</td>
                <td>
                    <input type='button' value='Редактировать' onClick={this.editClient} />
                </td>
                <td>
                    <input type='button' value='Удалить' onClick={this.deleteClient} />
                </td>
            </tr>
        );

    }
}

export default MobileClient;