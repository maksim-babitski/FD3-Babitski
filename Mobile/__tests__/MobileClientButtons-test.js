"use strict";

import React from 'react';
import renderer from 'react-test-renderer';

import MobileClient from '../components/MobileClient';

let client = {id:1, lastName: "Иванов", name: "Иван", middleName: "Иванович", balance:200};

test('работа кнопки Редактировать', () => {

  const component = renderer.create(
    <MobileClient info={client} />
  );

  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  const buttonAllElem = component.root.findByProps({ value: "Редактировать" });
  
  buttonAllElem.props.onClick();

  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  buttonAllElem.props.onClick();
  
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('работа кнопки Удалить', () => {

  const component = renderer.create(
    <MobileClient info={client} />
  );

  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  const buttonAllElem = component.root.findByProps({ value: "Удалить" });
  
  buttonAllElem.props.onClick();

  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  buttonAllElem.props.onClick();
  
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
});