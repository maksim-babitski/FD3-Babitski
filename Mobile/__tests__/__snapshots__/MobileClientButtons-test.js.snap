// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа кнопки Редактировать 1`] = `
<tr
  className="MobileClient"
>
  <td>
    Иванов
  </td>
  <td>
    Иван
  </td>
  <td>
    Иванович
  </td>
  <td>
    200
  </td>
  <td
    style={
      Object {
        "background": "green",
      }
    }
  >
    active
  </td>
  <td>
    <input
      onClick={[Function]}
      type="button"
      value="Редактировать"
    />
  </td>
  <td>
    <input
      onClick={[Function]}
      type="button"
      value="Удалить"
    />
  </td>
</tr>
`;

exports[`работа кнопки Редактировать 2`] = `
<tr
  className="MobileClient"
>
  <td>
    Иванов
  </td>
  <td>
    Иван
  </td>
  <td>
    Иванович
  </td>
  <td>
    200
  </td>
  <td
    style={
      Object {
        "background": "green",
      }
    }
  >
    active
  </td>
  <td>
    <input
      onClick={[Function]}
      type="button"
      value="Редактировать"
    />
  </td>
  <td>
    <input
      onClick={[Function]}
      type="button"
      value="Удалить"
    />
  </td>
</tr>
`;

exports[`работа кнопки Редактировать 3`] = `
<tr
  className="MobileClient"
>
  <td>
    Иванов
  </td>
  <td>
    Иван
  </td>
  <td>
    Иванович
  </td>
  <td>
    200
  </td>
  <td
    style={
      Object {
        "background": "green",
      }
    }
  >
    active
  </td>
  <td>
    <input
      onClick={[Function]}
      type="button"
      value="Редактировать"
    />
  </td>
  <td>
    <input
      onClick={[Function]}
      type="button"
      value="Удалить"
    />
  </td>
</tr>
`;

exports[`работа кнопки Удалить 1`] = `
<tr
  className="MobileClient"
>
  <td>
    Иванов
  </td>
  <td>
    Иван
  </td>
  <td>
    Иванович
  </td>
  <td>
    200
  </td>
  <td
    style={
      Object {
        "background": "green",
      }
    }
  >
    active
  </td>
  <td>
    <input
      onClick={[Function]}
      type="button"
      value="Редактировать"
    />
  </td>
  <td>
    <input
      onClick={[Function]}
      type="button"
      value="Удалить"
    />
  </td>
</tr>
`;

exports[`работа кнопки Удалить 2`] = `
<tr
  className="MobileClient"
>
  <td>
    Иванов
  </td>
  <td>
    Иван
  </td>
  <td>
    Иванович
  </td>
  <td>
    200
  </td>
  <td
    style={
      Object {
        "background": "green",
      }
    }
  >
    active
  </td>
  <td>
    <input
      onClick={[Function]}
      type="button"
      value="Редактировать"
    />
  </td>
  <td>
    <input
      onClick={[Function]}
      type="button"
      value="Удалить"
    />
  </td>
</tr>
`;

exports[`работа кнопки Удалить 3`] = `
<tr
  className="MobileClient"
>
  <td>
    Иванов
  </td>
  <td>
    Иван
  </td>
  <td>
    Иванович
  </td>
  <td>
    200
  </td>
  <td
    style={
      Object {
        "background": "green",
      }
    }
  >
    active
  </td>
  <td>
    <input
      onClick={[Function]}
      type="button"
      value="Редактировать"
    />
  </td>
  <td>
    <input
      onClick={[Function]}
      type="button"
      value="Удалить"
    />
  </td>
</tr>
`;
